newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js

```diff
@@ -225,6 +227,21 @@ export default class EventsBasedBehaviorPropertiesEditor extends React.Component
                           click: () => this._moveProperty(i, i + 1),
                           enabled: i + 1 < properties.getCount(),
                         },
+                        {
+                          label: i18n._(t`Generate functions`),
+                          click: () =>
```

Comment:
What happens if you click twice on this option? There are identical functions that are created?

Reply:
You can't as it gets disabled and there is also a check in the method to avoid overwriting existing functions.

Reply:
No, it's disabled if one of them exist. I hesitated to make something more intelligent (as the existing function could be of the wrong type and break the generated one that depends on it) but it seems too complicated and hard to understand for users so I kept it simple.